app/server/endpoint/src/main/java/io/syndesis/server/endpoint/v1/handler/continuousdelivery/ContinuousDeliveryHandler.java

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.syndesis.server.endpoint.v1.handler.continuousdelivery;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.List;
+import java.util.Map;
+import javax.validation.constraints.NotNull;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.FormParam;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.SecurityContext;
+import javax.ws.rs.core.StreamingOutput;
+
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiParam;
+import io.syndesis.common.model.connection.ConnectionOverview;
+import io.syndesis.common.model.integration.ContinuousDeliveryEnvironment;
+import io.syndesis.common.model.integration.ContinuousDeliveryImportResults;
+import io.syndesis.common.model.integration.IntegrationDeployment;
+import io.syndesis.common.model.monitoring.IntegrationDeploymentStateDetails;
+
+@Api("continuous-delivery")
```

Comment:
Continuous delivery is a practice, this feature might enable that practice but it might be also used for different use cases. Naming this continuous delivery is a mistake IMHO.

Reply:
Yeah, I was looking from it inside out. I am inclined to just call it something like `SyndesisPublicAPI` for now. We can always refactor that one interface when the number of endpoints grows in the future.

Reply:
Renamed to `PublicApiHandler`.